### PR TITLE
Attach a tag when registering a model with `models:/{model_id}`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13701?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13701/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13701
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Title

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/tracking/_model_registry/fluent.py [hunk 1]
```diff
@@ -126,6 +127,19 @@ def _register_model(
...
+            model_id,
+            {
+                mlflow_tags.MLFLOW_REGISTERED_MODEL: json.dumps(
+                    {
+                        "name": create_version_response.name,
+                        "version": create_version_response.version,
+                    }
+                )
+            },
+        )
```
Comment (dbczumar) [inline mlflow/tracking/_model_registry/fluent.py:142]: @harupy If there's already a tag present for this logged model, can we append a new entry to the list? There may be multiple model versions registered from the same logged model

File: mlflow/utils/mlflow_tags.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 MLFLOW_GIT_BRANCH = "mlflow.source.git.branch"
 MLFLOW_GIT_REPO_URL = "mlflow.source.git.repoURL"
 MLFLOW_LOGGED_MODELS = "mlflow.log-model.history"
+MLFLOW_REGISTERED_MODELS = "mlflow.registeredModels"
```
Comment (dbczumar) [inline mlflow/utils/mlflow_tags.py:24]:
```suggestion
MLFLOW_MODEL_VERSIONS = "mlflow.modelVersions"
```

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/13701/files#r1833617300 is addressed :)

File: examples/mlflow-3/register_model.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    model_info = mlflow.sklearn.log_model(model, "model")
+
+mlflow.register_model(model_info.model_uri, name="model")
+m = mlflow.get_logged_model(model_info.model_id)
+assert len(json.loads(m.tags["mlflow.registeredModels"])) == 1
+print(m.tags)
+
+mlflow.register_model(model_info.model_uri, name="hello")
+m = mlflow.get_logged_model(model_info.model_id)
+assert len(json.loads(m.tags["mlflow.registeredModels"])) == 2
```
Comment (harupy) [inline examples/mlflow-3/register_model.py:18]:
```suggestion
assert len(json.loads(m.tags["mlflow.modelVersions"])) == 2
```

File: examples/mlflow-3/register_model.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import mlflow
+
+with mlflow.start_run():
+    model = LinearRegression().fit([[1], [2]], [3, 4])
+    model_info = mlflow.sklearn.log_model(model, "model")
+
+mlflow.register_model(model_info.model_uri, name="model")
+m = mlflow.get_logged_model(model_info.model_id)
+assert len(json.loads(m.tags["mlflow.registeredModels"])) == 1
```
Comment (harupy) [inline examples/mlflow-3/register_model.py:13]:
```suggestion
assert len(json.loads(m.tags["mlflow.modelVersions"])) == 1
```

File: mlflow/tracking/_model_registry/fluent.py [hunk 5]
```diff
@@ -126,6 +127,23 @@ def _register_model(
...
+
+    if model_id := _parse_model_id_if_present(model_uri):
+        new_value = [
+            {
+                "name": create_version_response.name,
+                "version": create_version_response.version,
+            }
+        ]
+        model = client.get_logged_model(model_id)
+        if existing_value := model.tags.get(mlflow_tags.MLFLOW_REGISTERED_MODELS):
```
Comment (harupy) [inline mlflow/tracking/_model_registry/fluent.py:139]:
```suggestion
        if existing_value := model.tags.get(mlflow_tags.MLFLOW_MODEL_VERSIONS):
```

File: mlflow/tracking/_model_registry/fluent.py [hunk 6]
```diff
@@ -126,6 +127,23 @@ def _register_model(
...
+            new_value = json.loads(existing_value) + new_value
+
+        client.set_logged_model_tags(
+            model_id,
+            {mlflow_tags.MLFLOW_REGISTERED_MODELS: json.dumps(new_value)},
```
Comment (harupy) [inline mlflow/tracking/_model_registry/fluent.py:144]:
```suggestion
            {mlflow_tags.MLFLOW_MODEL_VERSIONS: json.dumps(new_value)},
```

**Pull request merged**